lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -552,23 +554,39 @@ def _get_subsection_grade_and_verify(self, all_earned, all_possible, graded_earn
         self.assertEqual(grade.graded_total.possible, graded_possible)
 
     @patch('crum.get_current_request')
-    def test_delete_student_state(self, _crum_mock):
+    @patch('lms.djangoapps.instructor.enrollment.tracker')
+    @patch('track.request_id_utils.uuid4')
+    def test_delete_student_state(self, mock_uuid, tracker_mock, _crum_mock):
+        permanent_uuid = uuid4()
+        mock_uuid.return_value = permanent_uuid
         problem_location = self.problem.location
         self._get_subsection_grade_and_verify(0, 1, 0, 1)
         answer_problem(course=self.course, request=self.request, problem=self.problem, score=1, max_value=1)
         self._get_subsection_grade_and_verify(1, 1, 1, 1)
-
         # Delete student state using the instructor dash
         reset_student_attempts(
             self.course.id,
             self.user,
             problem_location,
-            requesting_user=self.user,
+            requesting_user=self.instructor,
             delete_module=True,
         )
         # Verify that the student's grades are reset
         self._get_subsection_grade_and_verify(0, 1, 0, 1)
 
+        # Verify that the correct grading event is emitted
+        tracker_mock.emit.assert_called_with(
```

Comment:
Isn't this already verified in grades/tests/integration/test_events.py?  Why test again here?